pthelma/enhydris_api.py

```diff
@@ -70,17 +71,17 @@ def delete_model(base_url, session_cookies, model, id):
         raise requests.exceptions.HTTPError()
 
 
-def read_tsdata(base_url, session_cookies, ts):
-    r = requests.get(base_url + 'api/tsdata/{0}/'.format(ts.id),
+def read_tsdata(base_url, session_cookies, ts_id):
+    r = requests.get(base_url + 'api/tsdata/{0}/'.format(ts_id),
                      cookies=session_cookies)
     r.raise_for_status()
-    ts.read(StringIO(r.text))
+    return pd.read_csv(StringIO(r.text), header=None, parse_dates=True, index_col=0)
 
 
-def post_tsdata(base_url, session_cookies, timeseries):
+def post_tsdata(base_url, session_cookies, timeseries_id, ts):
```

Comment:
Please consider to change the `post_tsdata` to `put_data`. 

The api says "Take a timeseries id and return the actual timeseries data to the client, or update a time series with new records.". You are not creating to "post" it but getting or update that in terms of API is "get or put"

<img width="810" alt="screenshot 2018-10-19 21 59 02" src="https://user-images.githubusercontent.com/5064612/47238235-346a7480-d3ea-11e8-8771-9a326ca26825.png">

Reply:
This doesn't have to do with this ticket (`enhydris_api` is an api client for Enhydris), but with the API of Enhydris. And it's a big discussion, because it's not really clear how we should distinguish between POST, PUT and PATCH for time series data (it depends on whether the "resource" is the time series as a whole or a record of the time series).

Besides, `enhydris_api` still needs a hell of stuff to become sane.